fontforgeexe/fontview.c

```diff
@@ -1029,25 +1030,193 @@ char *GetPostScriptFontName(char *dir, int mult) {
 return( temp );
 }
 
+struct gfc_data {
+    int done;
+    int ret;
+    GGadget *gfc;
+    GGadget *ignore_invalid;
+    SplineFont *sf;
+    EncMap *map;
+};
+
+static int e_h(GWindow gw, GEvent *event) {
+    if ( event->type==et_close ) {
+    struct gfc_data *d = GDrawGetUserData(gw);
+    d->done = true;
+    d->ret = false;
+    } else if ( event->type==et_char ) {
+return( false );
+    } else if ( event->type==et_map ) {
+    GDrawRaise(gw);
+    } else if ( event->type==et_mousemove ||
+        (event->type==et_mousedown && event->u.mouse.button==3 )) {
+    struct gfc_data *d = GDrawGetUserData(gw);
+    GFileChooserPopupCheck(d->gfc,event);
+    } else if (( event->type==et_mouseup || event->type==et_mousedown ) &&
+        (event->u.mouse.button>=4 && event->u.mouse.button<=7) ) {
+    struct gfc_data *d = GDrawGetUserData(gw);
+return( GGadgetDispatchEvent((GGadget *) (d->gfc),event));
+    }
+return( true );
+}
+
+static int FEAT_ImportOk(GGadget *g, GEvent *e) {
+    if ( e->type==et_controlevent && e->u.control.subtype==et_buttonactivate ) {
+        struct gfc_data *d = GDrawGetUserData(GGadgetGetWindow(g));
+        unichar_t *ret = GGadgetGetTitle(d->gfc);
+        char *temp = u2def_copy(ret);
+        int ignore_invalid = GGadgetIsChecked(d->ignore_invalid);
+        GGadget *tf;
+
+        GFileChooserGetChildren(d->gfc,NULL,NULL,&tf);
+        if ( *_GGadgetGetTitle(tf)=='\0' )
+            return( true );
+        GDrawSetCursor(GGadgetGetWindow(g),ct_watch);
+
+        if ( !LoadKerningDataFromMetricsFile(d->sf,temp,d->map,ignore_invalid))
+            ff_post_error(_("Load of Kerning Metrics Failed"),_("Failed to load kern data from %s"), temp);
+        d->done = true;
+
+        GDrawSetCursor(GGadgetGetWindow(g),ct_pointer);
+        free(ret); free(temp);
+    }
+    return( true );
+
+}
+
+static int FEAT_Cancel(GGadget *g, GEvent *e) {
+    if ( e->type==et_controlevent && e->u.control.subtype == et_buttonactivate ) {
+        struct gfc_data *d = GDrawGetUserData(GGadgetGetWindow(g));
+        d->done = true;
+        d->ret = false;
+    }
+    return( true );
+}
+
 void MergeKernInfo(SplineFont *sf,EncMap *map) {
 #ifndef __Mac
-    static char wild[] = "*.{afm,tfm,ofm,pfm,bin,hqx,dfont,feature,feat,fea}";
-    static char wild2[] = "*.{afm,amfm,tfm,ofm,pfm,bin,hqx,dfont,feature,feat,fea}";
+    static unichar_t wild[] = {'*','.','{','a','f','m',',','t','f','m',',','o','f','m',',','p','f','m',',','b','i','n',',','h','q','x',',','d','f','o','n','t',',','f','e','a','t','u','r','e',',','f','e','a','t',',','f','e','a','}'
+};
+    static unichar_t wild2[] = {'*','.','{','a','f','m',',','a','m','f','m',',','t','f','m',',','o','f','m',',','p','f','m',',','b','i','n',',','h','q','x',',','d','f','o','n','t',',','f','e','a','t','u','r','e',',','f','e','a','t',',','f','e','a','}'
+};
```

Comment:
Why did this need to change? It seems a step backwards in terms of readability.

Reply:
I don’t like it either, but that's that `GFileChooserSetFilterText` expects as parameter (otherwise I got a crash).

Reply:
Tried a couple of things: passing `wild/wild2` as `static char array` results in segfault. Changing `char wiild[]` to `unichar_t wild[]` fails with compile error `cannot initialize array of ‘unsigned int’ from a string literal with type array of ‘char’`. Any ideas?

Reply:
Oh, I see. Let's leave it for now. [This has come up before](https://github.com/fontforge/fontforge/issues/3512#issuecomment-509465401), and we haven't yet decided whether or not C11 string literals are OK. For the record, I was wrong there, in this case it would be `static unichar_t wild[] = U"..."`, but you get the point :-)

If we decide it's OK, I'll convert these all at once, but that's something I can handle at a later date.